src/Juvix/Compiler/Nockma/Translation/FromTree.hs

```diff
@@ -1092,26 +1070,25 @@ callFunWithArgs fun args = do
   let p' = fpath ++ closurePath FunCode
   return (opCall ("callFun-" <> fname) p' newSubject)
 
-callClosure :: (Members '[Reader CompilerCtx] r) => TempRef -> [Term Natural] -> Sem r (Term Natural)
-callClosure ref newArgs = do
+callClosure :: (Member (Reader CompilerCtx) r) => TempRef -> [Term Natural] -> Sem r (Term Natural)
+callClosure ref args = do
   -- We never call a closure with zero arguments: if there are no arguments then
   -- there is no application and the closure is just returned. This differs from
   -- the behaviour with calls to known functions which may have zero arguments.
-  massert (not (null newArgs))
+  massert (not (null args))
   closure <- addressTempRef ref
-  let oldArgsNum = getClosureField ClosureArgsNum closure
-      oldArgs = getClosureField ClosureArgs closure
-  allArgs <- appendToTuple oldArgs oldArgsNum (foldTermsOrNil newArgs)
-  newSubject <- replaceSubject $ \case
-    FunCode -> Just (getClosureField FunCode closure)
-    ArgsTuple -> Just allArgs
-    FunctionsLibrary -> Nothing
-    StandardLibrary -> Nothing
-    ClosureArgs -> Nothing
-    ClosureTotalArgsNum -> Nothing
-    ClosureArgsNum -> Nothing
-    AnomaGetOrder -> Nothing
-  return (opCall "callClosure" (closurePath FunCode) newSubject)
+  let closure' = OpReplace # (closurePath ArgsTuple # foldTermsOrQuotedNil args) # closure
+  return (opCall "callClosure" (closurePath FunCode) closure')
+
+curryClosure :: Term Natural -> [Term Natural] -> Term Natural -> Sem r (Term Natural)
+curryClosure f args newArity = do
+  let args' = (foldTerms (nonEmpty' $ map (\x -> (OpQuote # OpQuote) # x) args <> [OpQuote # OpAddress # closurePath ArgsTuple]))
+  return . makeClosure $ \case
+    FunCode -> (OpQuote # OpCall) # (OpQuote # closurePath FunCode) # (OpQuote # OpReplace) # ((OpQuote # closurePath ArgsTuple) # args') # (OpQuote # OpQuote) # f
+    ArgsTuple -> OpQuote # nockNilTagged "argsTuple" -- We assume the arguments tuple is never accessed before being replaced by the caller.
+    ClosureRemainingArgsNum -> newArity
+    FunctionsLibrary -> OpQuote # functionsLibraryPlaceHolder
+    StandardLibrary -> OpQuote # stdlibPlaceHolder
```

Comment:
These fields don't need to be addresses to a the functions library and stdlib in the function wrapper? Is this because the function code is not evaluated in this context? Perhaps it's a good idea to add a comment?